mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -42,16 +42,17 @@
     @Requirement
     public org.codehaus.plexus.logging.Logger mavenLogger;
 
-    private final ChangeStorage changeStorage = new LocalChangeStorage(".");
+    private final ChangeStorage changeStorage = new LocalChangeStorage();
 
     private Configuration configuration;
 
     private Boolean skipExtensionInstallation;
 
     @Override
     public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
+        File projectDirectory = session.getCurrentProject().getModel().getProjectDirectory();
```

Comment:
This won't give you correct projectDir if you are running in some module  which again contains module.
e.g.
```xml
parent
 - config
   -- api
     --- pom.xml
   -- impl-base
      --- pom.xml
   -- spi
      --- pom.xml
```
```bash
cd parent && mvn clean test
```
This method will return `config/api` as projectDir. which is not expected. It should return `config` as basedir.

We should use something like
`File projectDirectory = session.getTopLevelProject().getModel().getProjectDirectory()`

Reply:
Good catch! I wouldn't notice this... Thx 👍